### PR TITLE
reuse python code from datadog_checks_base

### PR DESCRIPTION
**Pull request by ofek**
This removes all code that is also in https://github.com/DataDog/integrations-core/tree/master/datadog_checks_base by simply importing what is necessary.

### Motivation

Deduplication of logic: We often have to update code in both places at the same time.

depends on:
https://github.com/DataDog/integrations-core/pull/1561 :heavy_check_mark:
https://github.com/DataDog/integrations-core/pull/1562 :heavy_check_mark:
https://github.com/DataDog/integrations-core/pull/1565 :heavy_check_mark:
https://github.com/DataDog/integrations-core/pull/1566 :heavy_check_mark:

### CODE REVIEW

**Comment by hush-hush** [comment]
@ofek: could you explain a little more what this PR does and why ?
Thanks

File: tasks/go.py [hunk 1]
```diff
@@ -174,16 +174,31 @@ def misspell(ctx, targets):
         print("misspell found no issues")
 
 @task
-def deps(ctx):
+def deps(ctx, no_checks=False, core_dir=None, verbose=False):
```
Comment (olivielpeau) [inline tasks/go.py]: Can you document how a developer is supposed to use these options when setting up their dev environment? Could be documented in the command help, might be worth updating the `docs/dev/` if applicable.

File: tasks/go.py [hunk 2]
```diff
@@ -174,16 +174,31 @@ def misspell(ctx, targets):
...
+    ctx.run('go get{} -u github.com/golang/dep/cmd/dep'.format(verbosity))
+    ctx.run('go get{} -u github.com/golang/lint/golint'.format(verbosity))
+    ctx.run('go get{} -u github.com/fzipp/gocyclo'.format(verbosity))
+    ctx.run('go get{} -u github.com/gordonklaus/ineffassign'.format(verbosity))
+    ctx.run('go get{} -u github.com/client9/misspell/cmd/misspell'.format(verbosity))
+    ctx.run('dep ensure{}'.format(verbosity))
+
+    if not no_checks:
+        verbosity = 'v' if verbose else 'q'
+        core_dir = core_dir or os.getenv('DD_CORE_DIR')
```
Comment (olivielpeau) [inline tasks/go.py]: `DD_CORE_DIR` should be documented

**Review by olivielpeau — COMMENTED** [review]
Made a few inline comments, overall LGTM.

A few general comments/questions:
1. can you squash your commits?
2. can you add a `reno` release note with a `deprecations` section mentioning that these python import paths should be changed (in user-provided custom checks)?
3. Is there a plan to do a similar change in Agent5? If not I think we should also list this deprecation in `docs/agent/changes.md`

File: cmd/agent/dist/checks/network_checks.py [hunk 3]
```diff
@@ -2,59 +2,4 @@
...
-                    self.report_as_service_check(sc_name, status, instance, msg)
-
-    def _check(self, instance):
-        """This function should be implemented by inherited classes"""
-        raise NotImplementedError
-
-    def report_as_service_check(self, sc_name, status, instance, msg=None):
-        """This function should be implemented by inherited classes"""
-        raise NotImplementedError
+from datadog_checks.checks import NetworkCheck, Status
```
[the remark this answers is not in the document]
Comment (ofek) [reply cmd/agent/dist/checks/network_checks.py]: It's not used anywhere. Shall I add it anyway?

File: cmd/agent/dist/checks/prometheus_check/__init__.py [hunk 4]
```diff
@@ -1,6 +1,4 @@
 # (C) Datadog, Inc. 2016
 # All rights reserved
 # Licensed under Simplified BSD License (see LICENSE)
-
-from prometheus_check import *
-from . functions import parse_metric_family  # noqa: F401
+from datadog_checks.checks.prometheus import PrometheusCheck
```
[the remark this answers is not in the document]
Comment (ofek) [reply cmd/agent/dist/checks/prometheus_check/__init__.py]: They are not used anywhere. Shall I add them anyway?

**Comment by olivielpeau** [comment]
@ofek about the legacy imports that I mentioned in my review: the rationale is to continue supporting the old import paths for custom checks that may still be using them, so I think we should do that for every class/function that could reasonably be imported from a custom check.

Ideally, once all the integrations-core checks use the new import paths, we'd log a deprecation warning when an old import path is used so that it's clear to users that they need to update their custom checks' code.

File: releasenotes/notes/remove-duplicated-python-code-d4e1e29f420206cd.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+---
+deprecations:
+  - |
+    The core Agent check Python code is no longer duplicated here and is instead
+    pulled from integrations-core. The code now resides in the `datadog_checks`
+    namespace, though the old `checks`, `utils`, etc. paths are still supported.
```
Comment (olivielpeau) [inline releasenotes/notes/remove-duplicated-python-code-d4e1e29f420206cd.yaml]: can you add something like: `Please update your custom checks accordingly` , and add a link to the relevant documentation?

File: docs/dev/agent_dev_env.md [hunk 6]
```diff
@@ -27,6 +27,40 @@ You must install [go](https://golang.org/doc/install) version 1.9.2 or above. Ma
 sure that `$GOPATH/bin` is in your `$PATH` otherwise Invoke cannot use any
 additional tool it might need.
 
+### Python
```
Comment (olivielpeau) [inline docs/dev/agent_dev_env.md]: this `Python` section mostly talks about installing the dev versions of python needed to build the agent against. So I think it should be left below in the `System or Embedded` section.

Here, you could simply mention in the `Invoke` section that a standard python+pip install is required for development maybe? What do you think?

**Review by olivielpeau — DISMISSED** [review]
LGTM except one nit on the docs

**Review by olivielpeau — APPROVED** [review]
Nice! 👌

**Pull request merged**